dask/dataframe/core.py

```diff
@@ -340,6 +341,26 @@ def __array__(self, dtype=None, **kwargs):
     def __array_wrap__(self, array, context=None):
         raise NotImplementedError
 
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        out = kwargs.get('out', ())
+        for x in inputs + out:
+            if not isinstance(x, (Number, Scalar, _Frame, Array,
+                                  pd.DataFrame, pd.Series, pd.Index)):
```

Comment:
Maybe include `np.ndarray` in this list?

Reply:
@shoyer sure. Currently ndarray is not handled as expected in ufunc wrapper for dataframes. I will need to dig a bit more into it to get it work.

Reply:
We could also merge this now and handle np.ndarray in future work.  The decision for this is probably up to you @almaleksia .

Reply:
OK great, lets leave that for future work then